src/Factory/RepositoryContainer.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Cycle\Factory;
+
+use Closure;
+use Cycle\ORM\ORMInterface;
+use Cycle\ORM\RepositoryInterface;
+use Cycle\ORM\SchemaInterface;
+use Psr\Container\ContainerInterface;
+
+use Yiisoft\Factory\Exception\NotFoundException;
+use function is_string;
+
+final class RepositoryContainer implements ContainerInterface
+{
+    private ContainerInterface $rootContainer;
+    private ORMInterface $orm;
+    private array $roles = [];
+    private array $instances;
+    private bool $build = false;
+
+    public function __construct(ContainerInterface $rootContainer)
+    {
+        $this->rootContainer = $rootContainer;
+    }
+
+
+    public function get($id)
+    {
+        if (isset($this->instances[$id])) {
+            return $this->instances[$id];
+        }
+
+        if ($this->has($id)) {
+            return $this->instances[$id] = $this->makeRepository($this->roles[$id]);
+        }
+
+        throw new NotFoundException("Repository $id doesn't exist.");
```

Comment:
It is preferable to throw several friendly errors here:
- "ambiguous repository definition" when the repository is used by several roles;
- "the repository is not defined in the ORM Schema" 
- "the requested object is not a repository (doesn't implements RepositoryInterface)"